client/src/App.js

```diff
@@ -1,23 +1,63 @@
-import React from "react";
-import { ThemeProvider } from "@material-ui/core/styles";
-import { BrowserRouter, Route, Redirect } from "react-router-dom";
+import * as React from "react";
+import { ThemeProvider } from '@material-ui/styles';
+import { UserContext } from "./contexts/UserContextProvider"
 import CssBaseline from '@material-ui/core/CssBaseline';
-
-
+import { BrowserRouter, Route, Redirect, Switch, useHistory } from "react-router-dom";
+import { ProtectedRoute } from "./components/ProtectedRoute";
+import AuthService from "./services/AuthService"
 import { theme } from "./themes/theme";
 import LoginSignUp from "./pages/LoginSignUp";
+import Home from "./pages/Home";
+import Page from "./components/Page";
 
 function App() {
+	const user = React.useContext(UserContext);
+	const history = useHistory();
+
+
+    const DefaultRoutes = () => {
+      return (
+        <div>
+          <Page>
+            <Switch>
+                <ProtectedRoute exact path="/home" component={Home} />
+            </Switch>
+          </Page>
+
+        </div>
+      );
+	};
+	
+	React.useEffect(() => {
+      if(document.cookie.indexOf('token') === -1){
+                  AuthService.withCookie()
+                        .then((data) => {
+                            if (data.user) {
+								user.setProfile(data.user);
+                            }
+                        })
+                        .catch((error) => {
```

Comment:
Watch the formatting - are you using a linter?

Also, can we put this into the userContext instead?
I think rather than checking document.cookie, it's be cleaner to just call a validation function regardless of whether or not there's a cookie. By fully encapsulating that validation, it'll be easier to make updates in the future if we change how we're handling authentication

Reply:
No problem for now as I am not sure we need it after refactoring